src/server/game/World/World.cpp

```diff
@@ -3293,6 +3289,14 @@ CharacterInfo const* World::GetCharacterInfo(ObjectGuid const& guid) const
     return nullptr;
 }
 
+ObjectGuid World::GetCharacterGuidByName(std::string const& name) const
```

Comment:
Because rest of CharacterInfo is stored in world, if this should be moved to ObjectMgr then CharacterInfo should be as well.